apps/checkout-app/backend/payments/providers/adyen/index.ts

```diff
@@ -65,9 +74,146 @@ export const createAdyenPayment = async (
   return url;
 };
 
-export const verifyPayment = async (
+export const isNotificationDuplicate = async (
+  transactions: TransactionFragment[],
+  notificationItem: Types.notification.NotificationRequestItem
+) => {
+  const eventKeys = transactions
+    .map(({ events }) => events.map(createEventUniqueKey))
+    .flat();
+  const newEventKey = createEventUniqueKey({
+    reference: notificationItem.pspReference,
+    name: notificationItem.eventCode.toString(),
+  });
+
+  return eventKeys.includes(newEventKey);
+};
+
+export const getOrderId = async (
+  notification: Types.notification.NotificationRequestItem
+) => {
+  const { additionalData } = notification;
+  const paymentLinkId = additionalData?.paymentLinkId;
+
+  if (!paymentLinkId) {
+    return;
+  }
+
+  const { metadata } = await checkout.getPaymentLinks(paymentLinkId);
+
+  return metadata?.orderId;
+};
+
+export const getUpdatedTransactionData = (
```

Comment:
Not a fan of having functions inside a function with logic intertwined between. Maybe a separate file with export from this index would be nice if this function and it's sub functions are complicated enough to require such a structure

Reply:
These functions are closely related to that particular logic so I'd rather not abstract them away, they are mostly quite simple and I made them on purpose to avoid a very long a hard to read ternary operator